### DISCUSSION TITLE
How many bit widths does a bipolar data use in FINN hardware implementation?

### DISCUSSION THREAD

**Discussion by MFJI** [Xilinx/finn#847] [Q&A] [answered]
Hello all, in FINN implementation, there is a data format called bipolar (-1 or 1). Does anyone know how many hardware bit widths a bipolar data use in FINN hardware implementation? Is it 1 bit or 2 bits? Thank you in advance.

**Comment by auphelia** [comment, marked as answer]
Hi @MFJI,

In FINN we're working with 1-bit bipolar data. That means 0 represents -1 and 1 represents +1. It is more detailed described in the FINN paper and from the `tfc_end2end_example` Jupyter notebook: _Our example network is a quantized network with 1-bit bipolar (-1, +1 values) precision, and we want FINN to implement them as XNOR-popcount operations [as described in the original FINN paper](https://arxiv.org/pdf/1612.07119)._
